server.go

```diff
@@ -357,15 +395,29 @@ func Serve(opts *ServeConfig) {
 
 	logger.Debug("plugin address", "network", listener.Addr().Network(), "address", listener.Addr().String())
 
-	// Output the address and service name to stdout so that the client can bring it up.
-	fmt.Printf("%d|%d|%s|%s|%s|%s\n",
-		CoreProtocolVersion,
-		protoVersion,
-		listener.Addr().Network(),
-		listener.Addr().String(),
-		protoType,
-		serverCert)
-	os.Stdout.Sync()
+	// Output the address and service name to stdout so that the client can
+	// bring it up. In test mode, we don't do this because clients will
+	// attach via a reattach config.
+	if opts.Test == nil {
+		fmt.Printf("%d|%d|%s|%s|%s|%s\n",
+			CoreProtocolVersion,
+			protoVersion,
+			listener.Addr().Network(),
+			listener.Addr().String(),
+			protoType,
+			serverCert)
+		os.Stdout.Sync()
+	} else if ch := opts.Test.ReattachConfigCh; ch != nil {
+		// Send back the reattach config that can be used. This isn't
+		// quite ready if they connect immediately but the client should
+		// retry a few times.
+		ch <- &ReattachConfig{
+			Protocol: protoType,
+			Addr:     listener.Addr(),
+			Pid:      os.Getpid(),
+			Test:     true,
+		}
+	}
 
 	// Eat the interrupts
```

Comment:
It'd be super cool if we could disable this in test mode; right now, tests can't be cancelled or killed by keyboard shortcut, you need to look up the process number and kill it with the `kill` unix command. Given the control of the process isn't supposed to reside with the client anymore, I would expect the process to respond to the interrupt signal.

Reply:
Ah yep, good catch. Fixed.